src/main/java/org/fenixedu/academic/domain/student/personaldata/EducationLevelType.java

```diff
@@ -0,0 +1,61 @@
+package org.fenixedu.academic.domain.student.personaldata;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.academic.util.Bundle;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.bennu.core.i18n.BundleUtil;
+import org.fenixedu.commons.i18n.LocalizedString;
+
+public class EducationLevelType extends EducationLevelType_Base {
+    protected EducationLevelType() {
+        super();
+        setRootDomainObject(Bennu.getInstance());
+    }
+
+    public static EducationLevelType create(String code, LocalizedString name, boolean active, boolean isForStudent,
+            boolean isForStudentHousehold, boolean isForMobilityStudent, boolean isOther, boolean isPhDDegree,
+            boolean isSchoolLevelBasicCycle, boolean isHighSchoolOrEquivalent, boolean isHigherEducation)
+    {
+        if (findByCode(code).isPresent()) {
+            throw new DomainException(
+                    BundleUtil.getString(Bundle.APPLICATION, "error.EducationLevelType.code.already.exists", code));
+        }
+
+        EducationLevelType educationLevelType = new EducationLevelType();
+
+        educationLevelType.setCode(code);
+        educationLevelType.setName(name);
+        educationLevelType.setActive(active);
+        educationLevelType.setForStudent(isForStudent);
+        educationLevelType.setForStudentHousehold(isForStudentHousehold);
+        educationLevelType.setForMobilityStudent(isForMobilityStudent);
+        educationLevelType.setOther(isOther);
+        educationLevelType.setPhDDegree(isPhDDegree);
+        educationLevelType.setSchoolLevelBasicCycle(isSchoolLevelBasicCycle);
+        educationLevelType.setHighSchoolOrEquivalent(isHighSchoolOrEquivalent);
+        educationLevelType.setHigherEducation(isHigherEducation);
+
+        return educationLevelType;
+    }
+
+    public void delete() {
+        getDegreeClassificationsSet().clear();
+        setRootDomainObject(null);
+        super.deleteDomainObject();
+    }
+
```

Comment:
The `setCode(code)` method should be overriden in order to check if already exists another instance with the same code